File: cpp/questions/countNegatives.cpp
class Solution {
public:
    int countNegatives(vector<vector<int>>& grid) {
        // int ans = 0;
        // int rs = grid.size();
        // int cs = grid[0].size();
        // int r = 0;
        // int c = cs-1;
        // while (r<rs && grid[r][c]>=0) {
        //     r++;
        // }
        // while (r < rs) {
        //     while (c>=0 && grid[r][c]<0) {
        //         c--;
        //     }
        //     ans += cs - c -1;
        //     r++;
        // }
        // return ans;
        int ans = 0;
        int rb = grid.size()-1;
        int cr = grid[0].size()-1;
        int cl = cr;
        int ru = 0;
        while (ru<rb && grid[ru][c]>=0) {
            ru++;
        }
        while (ru<rb && grid[rb][0]<0) {
            rb--;
            ans += grid[0].size();
        }
        while (ru <= rb) {
            while (cl>=0 && grid[ru][cl]<0) {
                cl--;
            }
            ans += cr - cl;
            ru++;
        }
        return ans;
    }
};

File: python/questions/countNegatives.py
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        rd = len(grid)-1
        cr = len(grid[0])-1
        ru = 0
        cl = cr
        ans = 0
        while ru<=rd and grid[ru][cr]>=0:
            ru += 1
        while ru<=rd and grid[rd][0]<0:
            rd -= 1
            ans += cr+1
        while ru <= rd:
            while cl>=0 and grid[ru][cl]<0:
                cl -= 1
            ans += cr - cl
            ru += 1
        return ans
            
